tests/duty.test.ts

```diff
@@ -0,0 +1,170 @@
+  await collection.deleteMany({});
+});
+
+afterAll(async () => {
+  collection.deleteMany({});
+});
+
+describe('POST /duties', () => {
+  test('Should return code 201 and object if all info passed correctly', async () => {
+    const duty = createTestDuty();
+    const res = await request(server).post(`/duties`).send(duty);
+    expect(res.statusCode).toEqual(201);
+    expect(DutySchema.parse(res.body)).toEqual({
+      ...duty,
+      soldiers: [],
+    });
+  });
+
+  test('Should return code 400 if missing a field', async () => {
+    const duty = createTestDuty();
+    const res = await request(server)
+      .post(`/soldiers`)
+      .send(lodash.omit(duty, ['value']));
+    expect(res.statusCode).toEqual(400);
+  });
+});
+
+describe('GET /duties', () => {
+  test('Should return code 200 and return duties by id', async () => {
+    const duties = Array.from({ length: 3 }, createTestDutyWithId);
+    await collection.insertMany(duties);
+    const res = await request(server).get(`/duties/${duties[0]._id}`);
+    expect(res.statusCode).toEqual(200);
+    expect(DutySchema.parse(res.body)).toEqual(DutySchema.parse(duties[0]));
+  });
+
+  test('Should return code 200 and return all duties which have a certain constraint', async () => {
+    const duties = Array.from({ length: 3 }, createTestDutyWithId);
+    await collection.insertMany(duties);
+    const res = await request(server).get(
+      `/duties?constraints[]=${duties[0].constraints[0]}`,
+    );
+    expect(res.statusCode).toEqual(200);
+    expect(res.body).toEqual(
+      duties
+        .map((d) => {
+          return {
+            ...d,
+            _id: d._id.toString(),
+            time: {
+              start: d.time.start.toISOString(),
+              end: d.time.end.toISOString(),
+            },
+          };
+        })
+        .filter((e) => e.constraints.includes(duties[0].constraints[0])),
+    );
+  });
+
+  test('Should return code 404 when soldier does not exist', async () => {
+    const res = await request(server).get(
+      `/duties/${createTestDutyWithId()._id}`,
+    );
+    expect(res.statusCode).toEqual(404);
+  });
+});
+
+describe('DELETE /duties', () => {
+  test('Should return code 200, remove the duty from db, and return the deleted duty', async () => {
+    const duties = Array.from({ length: 3 }, createTestDutyWithId);
+    await collection.insertMany(duties);
+    const res = await request(server).delete(`/duties/${duties[0]._id}`);
+    expect(res.statusCode).toEqual(200);
+    expect(DutySchema.parse(res.body)).toEqual(DutySchema.parse(duties[0]));
+    const docs = await collection.find({}).toArray();
+    expect(docs).toEqual(duties.slice(1));
+  });
+
+  test('Should return code 404 when soldier does not exist', async () => {
+    const res = await request(server).delete(
+      `/duties/${createTestDutyWithId()._id}`,
+    );
+    expect(res.statusCode).toEqual(404);
+  });
+});
+
+describe('PATCH /duties', () => {
```

Comment:
What about testing for updating the existing value ?

Reply:
wdym? I'm testing for updating the existing location

Reply:
resolved